src/illuminatio/illuminatio_runner.py

```diff
@@ -31,52 +38,67 @@ def build_result_string(port, target, should_be_blocked, was_blocked):
+    """
+    run_times = {"overall": "error"}
+    if not os.path.exists(CASE_FILE_PATH):
+        raise RuntimeError("Could not find cases.yaml in %s!" % CASE_FILE_PATH)
     results, test_run_times = run_all_tests()
     namespace = None
     name = None
     try:
         namespace = os.environ["RUNNER_NAMESPACE"]
         name = os.environ["RUNNER_NAME"]
     except KeyError:
-        logger.error("Cannot store output to disk, as env vars are not set")
-    logger.debug("Output EnvVars: RUNNER_NAMESPACE=%s, RUNNER_NAME=%s", namespace, name)
+        LOGGER.error("Could not store output to ConfigMap, as env vars are not set")
+    LOGGER.debug("Output EnvVars: RUNNER_NAMESPACE=%s, RUNNER_NAME=%s", namespace, name)
     if namespace is not None and name is not None:
         store_results_to_cfg_map(results, namespace, name + "-results",
-                                 {"tests": test_run_times})
-    logger.info("Finished running tests. Results:")
-    logger.info(results)
+                                 {"overall": run_times["overall"], "tests": test_run_times})
+    LOGGER.info("Finished running tests. Results:")
+    LOGGER.info(results)
     # Sleep some time until container is killed. TODO: continuous mode ???
     time.sleep(60 * 60 * 24)
 
 
 def run_all_tests():
+    """
+    Runs all tests,
+    returns the results and measured execution times
+    """
     pods_on_node = [ConcreteClusterHost(p.metadata.namespace, p.metadata.name) for p in get_pods_on_node().items]
     results = {}
-    with open(case_file_path, "r") as yamlFile:
-        cases = yaml.safe_load(yamlFile)
-        logger.debug("Cases: " + str(cases))
+    with open(CASE_FILE_PATH, "r") as yaml_file:
+        cases = yaml.safe_load(yaml_file)
+        LOGGER.debug("Cases: %s", str(cases))
         test_runtimes = {}
-        all_from_pods = [Host.from_identifier(from_host_string) for from_host_string in cases]
-        from_pods_on_node = filter_from_hosts(all_from_pods, pods_on_node)
-        for from_pod in from_pods_on_node:
-            pod_identifier = from_pod.to_identifier()
-            results[pod_identifier], test_runtimes[pod_identifier] = run_tests_for_from_pod(from_pod, cases)
+        all_sender_pods = [Host.from_identifier(from_host_string) for from_host_string in cases]
+        sender_pods_on_node = get_pods_contained_in_both_lists(all_sender_pods, pods_on_node)
+        # execute tests for each sender pod
+        for sender_pod in sender_pods_on_node:
+            pod_identifier = sender_pod.to_identifier()
+            results[pod_identifier], test_runtimes[pod_identifier] = run_tests_for_sender_pod(sender_pod, cases)
     return results, test_runtimes
 
 
-def filter_from_hosts(from_hosts, pods_on_node):
-    from_hosts_on_node = [host for host in from_hosts if host_on_node(host, pods_on_node)]
-    return from_hosts_on_node
+def get_pods_contained_in_both_lists(first_pod_list, second_pod_list):
```

Comment:
Could you add here a todo? Probably we can be smarter then iterating multiple times over both lists which will take a pretty long time in large clusters e.g. converting one list into a dict and only iterate once over it. But that's a task in the future :)